storage-node/src/services/webApi/app.ts

```diff
@@ -38,6 +40,16 @@ export async function createApp(config: AppConfig): Promise<Express> {
 
       next()
     },
+
+    // Pre validate file upload params
+    (req: express.Request, res: express.Response<unknown, AppConfig>, next: NextFunction) => {
+      if (req.path === '/api/v1/files') {
+        validateUploadFileParams(req, res).catch((error) => sendResponseWithError(res, next, error, 'upload'))
+      }
+
+      next()
```

Comment:
As mentioned before, you need to make sure `next()` is called only once and only after all async jobs have finished
```suggestion
      if (req.path === '/api/v1/files') {
        validateUploadFileParams(req, res)
          .then(next)
          .catch((error) => sendResponseWithError(res, next, error, 'upload'))
      } else {
        next()
      }
```